lib/internal/process/next_tick.js

```diff
@@ -19,6 +19,52 @@ function setupNextTick() {
   const { kInit, kDestroy, kAsyncIdCounter } = async_wrap.constants;
   const { async_id_symbol, trigger_async_id_symbol } = async_wrap;
 
+  class FixedQueue {
+    constructor(size) {
+      this.size = size;
+      this.mask = size - 1;
+      this.btm = 0;
+      this.top = 0;
+      this.list = new Array(size);
+      this.next = null;
+    }
+
+    push(data) {
+      this.list[this.top] = data;
+      this.top = (this.top + 1) & this.mask;
+    }
+
+    shift() {
+      var next = this.list[this.btm];
+      if (next === undefined) return null;
+      this.list[this.btm] = undefined;
+      this.btm = (this.btm + 1) & this.mask;
+      return next;
+    }
+  }
+
+  var head = new FixedQueue(2048);
+  var tail = head;
+
+  function push(data) {
+    if (head.list[head.top] !== undefined)
+      head = head.next = new FixedQueue(head.size);
```

Comment:
I would just use a constant for `2048` and use that here. That way the constructor does not need the size entry.